packages/oidc-http-server-pages/src/create-static-pages.ts

```diff
@@ -1,146 +0,0 @@
-): string {
-  let result = `module.exports[${S(Component.name)}] = function(props) {
-    const actualProps = JSON.stringify(
-      Object.keys(props)
-        .filter(prop => ${S(
-          props
-        )}.includes(prop) && props[prop] != null).sort());`;
-  for (const availableProps of allSubsets(props)) {
-    const propsObject: Partial<Record<string, string>> = {};
-    const replacers: string[] = [];
-    for (const prop of availableProps) {
-      propsObject[prop] = placeholder(prop);
-      replacers.push(
-        `.replaceAll(${S(placeholder(prop))}, escapeHTML(props[${S(prop)}]))`
-      );
-    }
-
-    const markup = renderStylesToString(
-      renderToStaticMarkup(React.createElement(Component, propsObject))
-    );
-    const markupHash = createHash('sha256').update(markup).digest('hex');
-    markupTable[markupHash] = markup;
-
-    result += `if (actualProps === ${S(S(availableProps.sort()))}) {
-      return (getMarkup(${S(markupHash)})${replacers.join('')});
-    }`;
-  }
-
-  result += '}\n';
-  return result;
-}
-
-// Create CJS module with pages as exports.
-export function generateStaticPagesModule<PropNames extends string>(
-  components: [
-    React.FunctionComponent<Partial<Record<PropNames, string>>> & {
-      name: string;
-    },
-    string[]
-  ][]
-): string {
-  let result = `
-  'use strict';
-  function escapeHTML(str) {
-    return str.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#039;");
-  }
-  `;
-  const markupTable: Record<string, string> = {};
-  for (const component of components) {
-    result += generateStaticPage(...component, markupTable);
-  }
-
-  // At the time of writing, brotli compression results in a 96% size decrease
-  // in the resulting file. We decompress the markup table lazily, so that it
-  // is not loaded into memory when OIDC is not in use.
-  const compressedMarkupTable = brotliCompressSync(S(markupTable), {
-    params: {
-      [zlibConstants.BROTLI_PARAM_MODE]: zlibConstants.BROTLI_MODE_TEXT,
-      [zlibConstants.BROTLI_PARAM_QUALITY]: zlibConstants.BROTLI_MAX_QUALITY,
-    },
-  });
-  result += `
-  const markupTableSrc = ${S(compressedMarkupTable.toString('base64'))};
-  let markupTable;
-  function getMarkup(hash) {
-    if (markupTable === undefined) {
-      markupTable = JSON.parse(
-        require('zlib').brotliDecompressSync(
-          Buffer.from(markupTableSrc, 'base64')));
-    }
-    return markupTable[hash];
-  }
-  `;
-  return result;
-}
-
-if (require.main === module) {
-  // eslint-disable-next-line no-console
-  console.log(
-    generateStaticPagesModule([
-      [
-        OIDCErrorPage,
-        [
-          'error',
-          'errorDescription',
-          'errorURI',
-          'productDocsLink',
-          'productName',
-        ],
-      ],
-      [OIDCAcceptedPage, ['productDocsLink', 'productName']],
-      [OIDCNotFoundPage, ['productDocsLink', 'productName']],
-    ])
-  );
```

Comment:
Makes sense, I will look into this. Thank you!

Reply:
So, now we're creating two files -> one .gz and one .js that decompresses and exports the templates. I kept brotli for the latter, the file is now 12 KB. I hope it's okay that the files are created like this during the build. Previously I kept the "console.log solution" because I assumed there was a reason for it, such as permissions when the build happens in another context. Do you know if that was the case or not? @addaleax

Reply:
Also I kept the brotli settings, but I wonder if the MAX_QUALITY setting makes sense if we already use the MODE_TEXT. I don't know much about compression, but it seems weird that for text compression there would be different quality settings 🤔

Reply:
I think they're fairly independent settings – MODE_TEXT is just a hint to the compressor that it's compressing an UTF-8-encoded string and not arbitrary binary data, MAX_QUALITY means that it's okay to be fairly slow while compressing data in exchange for a smaller compressed output

Reply:
ah, I misunderstood what MAX_QUALITY means then